### PR TITLE
Send ports

### PR DESCRIPTION
**Pull request by BeaudanBrown**
Save the IP and port which is now being sent from lokid
Send the IP and port to clients as well

NOTE: Will need to update the integration tests to work with the change from "port" to "storage_port"
https://github.com/msgmaxim/swarm_migration_testing/pull/2

### CODE REVIEW

File: httpserver/http_connection.cpp [hunk 1]
```diff
@@ -546,6 +543,26 @@ bool connection_t::parse_header(const char* first, Args... args) {
     return parse_header(first) && parse_header(args...);
 }
 
+json connection_t::get_snodes_json_by_pk(const std::string& pubKey) {
+
+    std::vector<sn_record_t> nodes = service_node_.get_snodes_by_pk(pubKey);
```
Comment (sachaaaaa) [inline httpserver/http_connection.cpp]: `const`

File: httpserver/http_connection.cpp [hunk 2]
```diff
@@ -546,6 +543,26 @@ bool connection_t::parse_header(const char* first, Args... args) {
     return parse_header(first) && parse_header(args...);
 }
 
+json connection_t::get_snodes_json_by_pk(const std::string& pubKey) {
```
Comment (sachaaaaa) [inline httpserver/http_connection.cpp]: You can make this function const` as well, after constifying `service_node::get_snodes_by_pk` :)

File: httpserver/http_connection.cpp [hunk 3]
```diff
@@ -723,21 +726,7 @@ void connection_t::process_retrieve_all() {
...
-    for (const auto& sn : nodes) {
-#ifdef INTEGRATION_TEST
-        snodes.push_back(std::to_string(sn.port));
-#else
-        snodes.push_back(sn.address);
-#endif
-    }
-
-    res_body["snodes"] = snodes;
+    json res_body = get_snodes_json_by_pk(pubKey);
```
Comment (sachaaaaa) [inline httpserver/http_connection.cpp]: `const`

File: httpserver/http_connection.cpp (hunk 1 above)
Comment (BeaudanBrown) [reply httpserver/http_connection.cpp]: Done

File: httpserver/http_connection.cpp (hunk 3 above)
Comment (BeaudanBrown) [reply httpserver/http_connection.cpp]: Tru

File: httpserver/http_connection.cpp (hunk 2 above)
Comment (BeaudanBrown) [reply httpserver/http_connection.cpp]: Correct

File: httpserver/http_connection.cpp [hunk 4]
```diff
@@ -546,6 +543,26 @@ bool connection_t::parse_header(const char* first, Args... args) {
     return parse_header(first) && parse_header(args...);
 }
 
+json connection_t::get_snodes_json_by_pk(const std::string& pubKey) const {
```
Comment (msgmaxim) [inline httpserver/http_connection.cpp]: Should this be a freestanding function that accepts `std::vector<sn_record_t>` and returns a json object? I think it is better to keep the API lean.

Comment (BeaudanBrown) [reply httpserver/http_connection.cpp]: V tru

**Pull request merged**